### PR TITLE
soundtracker: init at 1.0.0.1

### PR DESCRIPTION
**Pull request by fgaz**
###### Motivation for this change

#81815

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

Could someone run a darwin build? From what I can tell it should work

### CODE REVIEW

**Comment by veprbl** [comment]
> Could someone run a darwin build?

Currently broken:
```
clang -DHAVE_CONFIG_H -I. -I..  -DLOCALEDIR=\"/nix/store/5qq22b4cnzlkj8kawllxigccpx6cjbqs-soundtracker-1.0.0.1/share/locale\"   -g -O2 -Wall -DG_DISABLE_DEPRECATED -DGTK_DISABLE_DEPRECATED -I/nix/store/mi5nn4652va8x85i6jkklkk9p9s6afns-gtk+-2.24.32-dev/include/gtk-2.0 -I/nix/store/ihkf4wswz6a9szn9c3rncksd4xzxm5vh-gtk+-2.24.32/lib/gtk-2.0/include -I/nix/store/cidsh5xxh4wvl10bgh3crh5kvs0wra75-glib-2.62.5-dev/include -I/nix/store/cidsh5xxh4wvl10bgh3crh5kvs0wra75-glib-2.62.5-dev/include/glib-2.0 -I/nix/store/z3qp77iysv2k6jar5hbsfy9qpwn6697g-glib-2.62.5/lib/glib-2.0/include -I/nix/store/01q8ks4swc8hv7q66i7mb7la411cva5p-cairo-1.16.0-dev/include/cairo -I/nix/store/m87gkyg396lwwjr1c9v9qbf4mkcr527z-freetype-2.10.1-dev/include/freetype2 -I/nix/store/m87gkyg396lwwjr1c9v9qbf4mkcr527z-freetype-2.10.1-dev/include -I/nix/store/lpgdvxzg5jjjfswcpwm85y61wc8r6m0n-pango-1.44.7-dev/include/pango-1.0 -I/nix/store/isnl2dvy12gfpcvz5647gxj4565z7w6q-harfbuzz-2.6.4-dev/include/harfbuzz -I/nix/store/1v28lzjn5zmiz692zcr5v7wrxn2dgcmy-gdk-pixbuf-2.40.0-dev/include/gdk-pixbuf-2.0 -I/nix/store/scdawvy2v3l1r6lfprgd6gxhg2x53i1a-atk-2.34.1-dev/include/atk-1.0 -I/nix/store/z1rirm1xpyr314wzamzwjycy6z404c0g-xorgproto-2019.1/include -I/nix/store/dq7s5gfs8mm9xkdmqydxy68ax2yan89k-libX11-1.6.8-dev/include -I/nix/store/mi5nn4652va8x85i6jkklkk9p9s6afns-gtk+-2.24.32-dev/include/gtk-2.0 -I/nix/store/ihkf4wswz6a9szn9c3rncksd4xzxm5vh-gtk+-2.24.32/lib/gtk-2.0/include -I/nix/store/cidsh5xxh4wvl10bgh3crh5kvs0wra75-glib-2.62.5-dev/include -I/nix/store/cidsh5xxh4wvl10bgh3crh5kvs0wra75-glib-2.62.5-dev/include/glib-2.0 -I/nix/store/z3qp77iysv2k6jar5hbsfy9qpwn6697g-glib-2.62.5/lib/glib-2.0/include -I/nix/store/01q8ks4swc8hv7q66i7mb7la411cva5p-cairo-1.16.0-dev/include/cairo -I/nix/store/m87gkyg396lwwjr1c9v9qbf4mkcr527z-freetype-2.10.1-dev/include/freetype2 -I/nix/store/m87gkyg396lwwjr1c9v9qbf4mkcr527z-freetype-2.10.1-dev/include -I/nix/store/lpgdvxzg5jjjfswcpwm85y61wc8r6m0n-pango-1.44.7-dev/include/pango-1.0 -I/nix/store/isnl2dvy12gfpcvz5647gxj4565z7w6q-harfbuzz-2.6.4-dev/include/harfbuzz -I/nix/store/1v28lzjn5zmiz692zcr5v7wrxn2dgcmy-gdk-pixbuf-2.40.0-dev/include/gdk-pixbuf-2.0 -I/nix/store/scdawvy2v3l1r6lfprgd6gxhg2x53i1a-atk-2.34.1-dev/include/atk-1.0 -I/nix/store/kfyib1jy0pdb9jpl096xgnn1fi3km9d6-goocanvas-1.0.0/include/goocanvas-1.0 -I/nix/store/8k9vgpr0mcgcz4rfzfficl03fw946pjl-jack2-1.9.14/include -I/nix/store/pd0i1bqhpp4wq4gkaggbp1rds2vk7d7n-SDL-1.2.15-dev/include/SDL -I/nix/store/pd0i1bqhpp4wq4gkaggbp1rds2vk7d7n-SDL-1.2.15-dev/include/SDL -I/nix/store/pd0i1bqhpp4wq4gkaggbp1rds2vk7d7n-SDL-1.2.15-dev/include/SDL -D_GNU_SOURCE=1 -D_THREAD_SAFE -c -o keys.o keys.c
keys.c:38:10: fatal error: 'gdk/gdkx.h' file not found
#include <gdk/gdkx.h>
         ^~~~~~~~~~~~
1 error generated.
make[3]: *** [Makefile:536: keys.o] Error 1
make[3]: Leaving directory '/private/var/folders/1f/2mj_ng1n2y3_wc59fgw3fn_r0000gn/T/nix-build-soundtracker-1.0.0.1.drv-0/soundtracker-1.0.0.1/app'
make[2]: *** [Makefile:552: all-recursive] Error 1
make[2]: Leaving directory '/private/var/folders/1f/2mj_ng1n2y3_wc59fgw3fn_r0000gn/T/nix-build-soundtracker-1.0.0.1.drv-0/soundtracker-1.0.0.1/app'
make[1]: *** [Makefile:623: all-recursive] Error 1
make[1]: Leaving directory '/private/var/folders/1f/2mj_ng1n2y3_wc59fgw3fn_r0000gn/T/nix-build-soundtracker-1.0.0.1.drv-0/soundtracker-1.0.0.1'
make: *** [Makefile:402: all] Error 2
```

File: pkgs/applications/audio/soundtracker/default.nix [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+, pkg-config
+, autoconf
+, gtk2
+, alsaLib
+, SDL
+, jack2
+, goocanvas # graphical envelope editing
+}:
+
+with stdenv.lib; stdenv.mkDerivation rec {
```
Comment (doronbehar) [inline pkgs/applications/audio/soundtracker/default.nix]:
```suggestion
stdenv.mkDerivation rec {
```

**Review by doronbehar — CHANGES_REQUESTED** [review]
Builds and runs fine for me (on NixOS).

File: pkgs/applications/audio/soundtracker/default.nix (hunk 1 above)
Comment (fgaz) [reply pkgs/applications/audio/soundtracker/default.nix]: Did you test this? I'm using `optional` from `stdenv.lib`

**Comment by fgaz** [comment]
Thank you @veprbl. I have no idea why that would happen (the `-I` is there), so I marked it as broken for darwin.

File: pkgs/applications/audio/soundtracker/default.nix (hunk 1 above)
Comment (doronbehar) [reply pkgs/applications/audio/soundtracker/default.nix]: Tested without this change. This should work (untested):

```suggestion
let inherit (stdenv.lib) optionals; in

stdenv.mkDerivation rec {
```

Comment (fgaz) [reply pkgs/applications/audio/soundtracker/default.nix]: In the end I just qualified `optional`, since I'm only using it one time

**Review by doronbehar — APPROVED** [review]
Thanks for your contribution.

**Pull request merged**